core/CronArchive/QueueConsumer.php

```diff
@@ -544,10 +546,14 @@ public function usableArchiveExists(array $invalidatedArchive)
 
         // if valid archive already exists, do not re-archive
         $minDateTimeProcessedUTC = Date::now()->subSeconds(Rules::getPeriodArchiveTimeToLiveDefault($periodLabel));
-        $archiveIdAndVisits = ArchiveSelector::getArchiveIdAndVisits($params, $minDateTimeProcessedUTC, $includeInvalidated = true);
+        $archiveIdAndVisits = ArchiveSelector::getArchiveIdAndVisits($params, $minDateTimeProcessedUTC, $includeInvalidated = false);
 
         $idArchive = $archiveIdAndVisits[0];
-        return !empty($idArchive);
+        if (empty($idArchive)) {
+            return false;
+        }
+
+        return Date::factory($archiveIdAndVisits[4])->getDatetime();
```

Comment:
Not really related to this PR: But should we maybe change `getArchiveIdAndVisits` to return something more meaningful. Without looking into the method it's hard to understand what the forth element of the returned array actual is. 
Personally I would prefer to have at least named array keys or an object (or class) with defined properties, that would make the code easier to read and to understand...